service/matching/matching_engine.go

```diff
@@ -1503,7 +1495,11 @@ func (e *matchingEngineImpl) redirectToVersionedQueueForAdd(
 	// Have to look up versioning data.
 	userData, userDataChanged, err := baseTqm.GetUserData(ctx)
 	if err != nil {
-		return taskQueue, userDataChanged, err
+		if errors.Is(err, errUserDataDisabled) && buildId == "" {
+			// When user data disabled, send "default" tasks to unversioned queue.
+			return taskQueue, nil, nil
```

Comment:
I think you'll still want to return the `userDataChanged` channel here instead of explicitly returning `nil` because user data may later be enabled (when we start periodically reading this config).

Reply:
Ah yeah, that's true